src/azul/plugins/repository/tdr_anvil/__init__.py

```diff
@@ -241,17 +244,31 @@ def _batch_uuid(self,
                               self.batch_uuid_version,
                               self.bundle_uuid_version)
 
+    def count_files(self, source: TDRSourceSpec) -> int:
+        if source.prefix is None:
+            prefix = ''
+        else:
+            prefix = source.prefix.common
+            assert not any(map(str.isupper, prefix)), source
```

Comment:
```suggestion
        prefix = '' if source.prefix is None else source.prefix.common
        assert prefix == prefix.lower(), source
```

My version is stronger because the assertion isn't conditional. Again, assertions are there to establish simple preconditions, and to communicate those to the reader. Yes, we know an empty string can't have upper case characters in it. But as I laid out in PL, humans make mistakes, so the next person modifying this code may not immediately recognize that additional dependency. My version clearly communicates that prefix is lower case past a certain point in the function. It's also shorter. That's why I prefer it.